### DISCUSSION TITLE
"Internal" Input arguments for an OPCUA mehtod

### DISCUSSION THREAD

**Discussion by PhilWillecke** [FreeOpcUa/opcua-asyncio#1214] [General]
Hi there, 

i build a server with some methods, this is working fine. 

Now I want to pass some extra (internal) input arguments to this function, then the function is call through the client (UaExpert in my case), is ther any way to do so.

Thanks in advance.
Phil

**Comment by schroeder-** [comment]
The best way is to provide two version of the function, because there are no optional arguments in opc ua methods. An other way is to use Arrays as optional arguments, because uaexpert provides an empty array as default value.

**Reply by PhilWillecke** [reply]
okay thanks for that idea.

I'm not sure if you got my plan right.
I Just want to call the Method form the Client (without sending input arguments from uaexpert to the server)
But the function needs some input argument from the server script. This is what ein try to handle. Your I idea is to call a second function from the first one, i'm rigth? 
Could work but maybe not the best way
